drivers/api/no_os_i2c.c

```diff
@@ -58,23 +64,62 @@ int32_t no_os_i2c_init(struct no_os_i2c_desc **desc,
 
 	if (!param->platform_ops->i2c_ops_init)
 		return -ENOSYS;
-
+	//Initializing BUS descriptor
+	if (i2c_table[param->device_id] == NULL) {
+		ret = no_os_i2cbus_init(param);
+		if (ret)
+			return ret;
+	}
+	//Initilize I2C descriptor
 	ret = param->platform_ops->i2c_ops_init(desc, param);
 	if (ret)
 		return ret;
-
+	(*desc)->bus = i2c_table[param->device_id];
 	(*desc)->platform_ops = param->platform_ops;
 
 	return 0;
 }
 
+/**
+ * @brief Initialize the i2c bus communication peripheral.
+ * @param param - The structure that containes the i2c bus parameters
+ * @return 0 in case of success, error code otherwise
+*/
+int32_t no_os_i2cbus_init(const struct no_os_i2c_init_param *param)
+{
+	i2c_table[param->device_id] = (struct no_os_i2cbus_desc *)no_os_calloc(1,
```

Comment:
`param->device_id` should be verified first. And maybe a macro for the `i2c_table`'s size instead of 8 would simplify thing a bit.